buffer/src/main/java/io/netty/buffer/SimpleLeakAwareCompositeByteBuf.java

```diff
@@ -16,35 +16,43 @@
 package io.netty.buffer;
 
 
-import io.netty.util.ResourceLeak;
+import io.netty.util.ResourceLeakTracker;
 
 import java.nio.ByteOrder;
 
 final class SimpleLeakAwareCompositeByteBuf extends WrappedCompositeByteBuf {
 
-    private final ResourceLeak leak;
+    private final ResourceLeakTracker<ByteBuf> leak;
 
-    SimpleLeakAwareCompositeByteBuf(CompositeByteBuf wrapped, ResourceLeak leak) {
+    SimpleLeakAwareCompositeByteBuf(CompositeByteBuf wrapped, ResourceLeakTracker<ByteBuf> leak) {
         super(wrapped);
         this.leak = leak;
     }
 
     @Override
     public boolean release() {
-        boolean deallocated = super.release();
-        if (deallocated) {
-            leak.close();
+        // Call unwrap() before just in case that super.release() will change the ByteBuf instance that is returned
+        // by ByteBuf.
+        ByteBuf unwrapped = unwrap();
+        if (super.release()) {
+            boolean closed = leak.close(unwrapped);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "closed". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)